### PR TITLE
Test case improvement for MKLDNN on Gluon

### PR DESCRIPTION
**Pull request by juliusshufan**
## Description ##
This PR is a "follow-up" of previously merged https://github.com/apache/incubator-mxnet/pull/10764 
In this PR, we are covering:
1. Refine the cases on nn.Conv2D and change the input shape to hit the MKLDNN code path (Done);
2. Adding more test cases cover other gluon layers, like BN, Dense/FC, Pooling etc. from the "MKLDNN-specialty" perspective. (WIP)
  
### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
All the changes is reflected by tests/python/mkl/test_mkldnn.py

## Comments ##
1. For the correctness check on gluon computation, it follows the design used by tests/python/unittest/test_gluon.py, and therefore, the helper functions defined in tests/python/unitest/common.py is also used.

### CODE REVIEW

**Comment by zheng-da** [comment]
could you go the C++ code to double check the input arrays of mkldnn convolution? to make sure that the arrays are reshaped MKLDNN arrays. I don't remember I handled this case.
